testnet/stacks-node/src/neon_node.rs

```diff
@@ -2402,24 +2409,26 @@ impl BlockMinerThread {
             self.config.is_mainnet(),
             &mut miners_stackerdb,
             &election_sortition,
-        ) {
-            warn!("Failed to send mock proposal to stackerdb: {:?}", &e);
-            return Ok(());
-        }
+        )
+        .map_err(|e| {
+            warn!("Failed to write mock proposal to stackerdb.");
+            e
+        })?;
```

Comment:
As of Rust 1.76 you can do this, which is a bit simpler

```suggestion
        .inspect_err(|_| warn!("Failed to write mock proposal to stackerdb."))?;
```